torch/testing/_internal/common_device_type.py

```diff
@@ -1395,13 +1395,13 @@ def efail_fn(slf, *args, **kwargs):
 
 
 class onlyOn:
-    def __init__(self, device_type):
+    def __init__(self, device_type: Union[str, list]):
         self.device_type = device_type
 
     def __call__(self, fn):
         @wraps(fn)
         def only_fn(slf, *args, **kwargs):
-            if self.device_type != slf.device_type:
+            if slf.device_type not in self.device_type:
```

Comment:
The `self.device_type` could be either a string literal or a list. If the `self.device_type` is a string, things get a little tricky. Because `not in` does not guarantee an exact match between two string literals.